### DISCUSSION TITLE
Island Mission completition status placeholder

### DISCUSSION THREAD

**Discussion by Davisiiiik** [BG-Software-LLC/SuperiorSkyblock2#1998] [Pending Feature Requests] [answered]
### Feature description:
Placeholder, that will return the completition status of any mission, for example:

`%superior_player_mission_<mission_name>%`
`%superior_player_mission_miner%`

So if player has completed mission (or if the mission is `island: true` mission and is completed on players island), the result will be `yes`, otherwise `no`.
Return value can be linked to `PLACEHOLDER_NO` and `PLACEHOLDER_YES` language options.

### Reasoning

It could come very handy, when you have many `required-missions` for another mission and you want to show to the player which missions they already completed.

**Comment by AquariamcDev** [comment]
This placeholder exists but it works slightly differently `%superior_island_missions_completed_<category>%` & `%superior_player_missions_completed_<category>%`. This will return the amount of missions that are complete for each category for example:
![image](https://github.com/user-attachments/assets/a208de01-ecf5-4314-ab5d-2d0e758476fd)
![image](https://github.com/user-attachments/assets/dcd90c02-bb5a-4e04-b287-3537115c8865)

**Reply by AlmostPanda** [reply]
Im a friend of Davisiiiik and we wanted the placeholder `%superior_player_mission_<mission_name>%` because we use the `required-missions` option for few missions. 
Sometimes players ask why some missions aren't progressing, so we have to tell them or add a lore that they have to complete first this and this mission to be able to progress.

**Reply by AquariamcDev** [reply]
> Im a friend of Davisiiiik and we wanted the placeholder `%superior_player_mission_<mission_name>%` because we use the `required-missions` option for few missions. 
> 
> Sometimes players ask why some missions aren't progressing, so we have to tell them or add a lore that they have to complete first this and this mission to be able to progress.

That's what I was looking for also but opted to go with the method I posted above, maybe @OmerBenGera can implement this method if it isn't too much hassle, it would be nice to have an option in the lang file to set the placeholder return values.

**Comment by OmerBenGera** [comment, marked as answer]
Added in latest dev build
Use `%superior_island_mission_status_<mission>%` to get Yes / No for whether you completed the mission.

**Reply by AlmostPanda** [reply]
works, thank you
